examples/payload_scan/payload_scan.c

```diff
@@ -0,0 +1,256 @@
+                                RTE_LOG(INFO, APP, "Print delay = %d\n", print_delay);
+                                break;
+
+                        case 'i':
+                                forward_on_match = 1;
+                                RTE_LOG(INFO, APP,
+                                        "Inverse mode enabled: packets dropped on string hit and forwarded on mismatch\n");
+                                break;
+                        case '?':
+                                usage(progname);
+                                if (optopt == 'p')
+                                        RTE_LOG(INFO, APP, "Option -%c requires an argument.\n", optopt);
+                                else if (isprint(optopt))
+                                        RTE_LOG(INFO, APP, "Unknown option `-%c'.\n", optopt);
+                                else
+                                        RTE_LOG(INFO, APP, "Unknown option character `\\x%x'.\n", optopt);
+                                return -1;
+                        default:
+                                usage(progname);
+                                return -1;
+                }
+        }
+
+        if (!dst_flag) {
+                RTE_LOG(INFO, APP, "Payload NF requires a destination NF with the -d flag.\n");
+                return -1;
+        }
+        if (!input_flag) {
+                RTE_LOG(INFO, APP, "Payload NF requires a search term string with the -s flag.\n");
+                return -1;
+        }
+        return optind;
+}
+
+/*
+ * This function displays stats. It uses ANSI terminal codes to clear
+ * screen when called. It is called from a single non-master
+ * thread in the server process, when the process is run with more
+ * than one lcore enabled.
+ */
+static void
+do_stats_display(struct onvm_nf_info *nf_info) {
+        const char clr[] = {27, '[', '2', 'J', '\0'};
+        const char topLeft[] = {27, '[', '1', ';', '1', 'H', '\0'};
+        struct onvm_pkt_stats *stats = (struct onvm_pkt_stats *) nf_info->data;
+
+        /* Clear screen and move to top left */
+        printf("%s%s", clr, topLeft);
+        printf("Search term: %s\n", search_term);
+        printf("Packets accepted: %lu\n", stats->pkt_accept);
+        printf("Packets dropped: %lu\n", stats->pkt_drop);
+        printf("Packets not IPv4: %lu\n", stats->pkt_not_ipv4);
+        printf("Packets not UDP/TCP: %lu\n", stats->pkt_not_tcp_udp);
+        printf("Packets total: %lu", stats->pkt_total);
+
+        printf("\n\n");
+}
+
+static int
+packet_handler(struct rte_mbuf *pkt, struct onvm_pkt_meta *meta,
+               __attribute__((unused)) struct onvm_nf_info *nf_info) {
+        int udp_pkt, tcp_pkt;
+        char search_match;
+        static uint32_t counter = 0;
+        uint8_t *pkt_data;
+        struct onvm_pkt_stats *stats = (struct onvm_pkt_stats *) nf_info->data;
+
+        if (++counter == print_delay) {
```

Comment:
replace counter with `stats->pkt_total`

Reply:
If I replace counter with stats->pkt_total, I'll have to either set pkt_total back to 0 (which would mess up the stats), or update the print_delay variable. I think the ++counter solution was cleaner. What do you think?

Reply:
Ah I see, yeah didn't think about that. Keep it :+1: